pe.c

```diff
@@ -772,8 +899,18 @@ handle_image (void *data, unsigned int datasize,
 			clear_attrs &= ~MEM_ATTR_X;
 		}
 		update_mem_attrs(addr, length, set_attrs, clear_attrs);
-	}
 
+		if ((Section->Characteristics & EFI_IMAGE_SCN_CNT_CODE ||
+		     Section->Characteristics & EFI_IMAGE_SCN_CNT_INITIALIZED_DATA) &&
+		    Section->Characteristics & EFI_IMAGE_SCN_MEM_READ) {
+			efi_status = cache_section(li, Section->Name, base, raw_length);
```

Comment:
The first parameter needs to be `li->ParentHandle` rather than `li`, as that's where LoadImage will save the handle passed by the caller, `li` is a handle local to shim

Reply:
scratch that, user error